src/structure.js

```diff
@@ -68,6 +77,54 @@ Structure.prototype.cursor = function (path) {
   return Cursor.from(self.current, path, changeListener);
 };
 
+Structure.prototype.reference = function (path) {
+  var self = this, pathId = pathString(path);
+  var listenerNs = self._pathListeners[pathId];
```

Comment:
I think the current solution is a better approach in terms of iterating over and executing listeners on swap.